recipes/Text_to_SQL/Text_to_SQL.ipynb

```diff
@@ -0,0 +1,537 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **Find the database file for the dataset. Make sure to give the correct path of this file below in order to fetch the schema.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The below code snippet connects to a SQLite database (social_media.db) using the SQLDatabase utility from the langchain_community package and retrieves its schema.\n",
+    "\n",
+    "**NOTE** - You can recreate the SQLite database (social_media.db) from CSV files `TwitterDataset/Tweet.csv`, `TwitterDataset/Location.csv`, `TwitterDataset/User.csv` by running Notebook `TwitterDataset/TwitterSyntheticDataGen.ipynb`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.utilities import SQLDatabase\n",
+    "\n",
+    "# Note: to run in Colab, you need to upload the TwitterDataset//social_media.db file in the repo to the Colab folder first.\n",
+    "db = SQLDatabase.from_uri(\"sqlite:///TwitterDataset//social_media.db\", sample_rows_in_table_info=0)\n",
+    "\n",
+    "def get_schema():\n",
+    "    return db.get_table_info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Retrieve the SQL schema that will be wrapped in the prompt that we pass to the LLM to have more context and generate accurate response. The schema contains 3 tables:\n",
+    "\n",
+    "**location table:** This table stores information about specific locations, including country, state, state code, and city. It is identified by the LocationID column.\n",
+    "\n",
+    "**tweet table:** This table contains information about individual tweets, such as the time of posting (Weekday, Hour, Day), language (Lang), whether it was a reshare (IsReshare), tweet statistics (e.g., Reach, RetweetCount, Likes), sentiment, and tweet content (Text). It references two foreign keys: LocationID to the location table and UserID to the user table.\n",
+    "\n",
+    "**user table:** This table holds user information, identified by UserID, and stores the user's gender."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "db_schema = get_schema()\n",
+    "print(db_schema)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **Provide your Replicate API Token**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Provide the API key you had created previously to access LLM hosted in the Replicate platform."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass, os\n",
```

Comment:
Delete this block- replace with the pattern shown here:
https://github.com/ibm-granite-community/granite-code-cookbook/blob/main/recipes/Getting_Started_with_Granite_Code.ipynb

Reply:
Done